host-controller/src/main/java/org/jboss/as/domain/controller/operations/coordination/ServerOperationResolver.java

```diff
@@ -538,13 +542,28 @@ private static ModelNode getDomainModel(OperationContext context) {
             } else if (SYSTEM_PROPERTY.equals(type)) {
                 String affectedGroup = address.getElement(0).getValue();
                 result = getServerSystemPropertyOperations(operation, address, Level.SERVER_GROUP, domain, affectedGroup, host);
-            } else if (DEPLOYMENT_OVERLAY.equals(type) && address.getLastElement().getKey().equals(DEPLOYMENT)) {
+            } else if (DEPLOYMENT_OVERLAY.equals(type)) {
+                ModelNode serverOp = operation.clone();
                 String groupName = address.getElement(0).getValue();
                 Set<ServerIdentity> servers = getServersForGroup(groupName, host, localHostName, serverProxies);
-                ModelNode serverOp = operation.clone();
-                PathAddress serverAddress = address.subAddress(1);
-                serverOp.get(OP_ADDR).set(serverAddress.toModelNode());
-                result = Collections.singletonMap(servers, serverOp);
+                if (address.getLastElement().getKey().equals(DEPLOYMENT)) {
+                    PathAddress serverAddress = address.subAddress(1);
+                    serverOp.get(OP_ADDR).set(serverAddress.toModelNode());
+                    result = Collections.singletonMap(servers, serverOp);
+                } else if (COMPOSITE.equals(operation.get(OP).asString())) { //This should be a redeploy affected transformed operation
```

Comment:
This needs more of a comment explaining how this would even happen as it's not something people will expect. Probably point people to AffectedDeploymentOverlay.